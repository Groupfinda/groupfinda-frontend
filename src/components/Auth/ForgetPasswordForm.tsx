import React, { useState } from "react";
import { Layout, Icon, Input, Button, Text } from "@ui-kitten/components";
import { StyleSheet } from "react-native";

type Props = {};

type ForgetPasswordVariables = {
  username: string;
  email: string;
};

type ReferencesType = {
  [key: string]: Input | null;
};

const ForgetPasswordForm: React.FC<Props> = () => {
  const [username, setUsername] = useState<string>("");
  const [email, setEmail] = useState<string>("");

  const references: ReferencesType = {};

  const onSubmit = (): void => {
    const variables: ForgetPasswordVariables = {
      username,
      email,
    };
    console.log(variables);
  };

  return (
    <Layout style={styles.containerStyle}>
      <Input
        autoCorrect={false}
        style={styles.usernameStyle}
        value={username}
        label="Username"
        placeholder="Username"
        accessoryLeft={(props) => <Icon {...props} name="person-outline" />}
        onChangeText={setUsername}
        onSubmitEditing={() => references.secondInput?.focus()}
        blurOnSubmit={false}
        textContentType="username"
      />
      <Input
        autoCorrect={false}
        value={email}
        label="Email"
        placeholder="Email"
        accessoryLeft={(props) => <Icon {...props} name="email-outline" />}
        onChangeText={setEmail}
        onSubmitEditing={onSubmit}
        ref={(ref) => (references.secondInput = ref)}
        textContentType="emailAddress"
      />
      <Button
        onPress={onSubmit}
        style={styles.buttonStyle}
        appearance="filled"
        status="primary"
      >
        Reset my password
      </Button>
    </Layout>
  );
};

const styles = StyleSheet.create({
  containerStyle: {
    alignSelf: "stretch",
  },
  usernameStyle: {
    marginBottom: 5,
  },
  buttonStyle: {
    marginTop: 40,
  },
});

export default ForgetPasswordForm;
