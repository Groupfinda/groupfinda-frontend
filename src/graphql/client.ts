import { ApolloClient } from "apollo-client";
import { createHttpLink } from "apollo-link-http";
import { InMemoryCache } from "apollo-boost";
import { setContext } from "apollo-link-context";
import { AsyncStorage } from "react-native";
import fetch from "isomorphic-fetch";

const authLink = setContext(async (_, { headers }) => {
  const token = await AsyncStorage.getItem("userToken");
  return {
    headers: {
      ...headers,
      authorization: token ? `bearer ${token}` : null,
    },
  };
});

const GRAPHQL_URL = `http://192.168.1.207:3002/graphql`;

const link = createHttpLink({
  uri: GRAPHQL_URL,
  fetch,
});

const client = new ApolloClient({
  cache: new InMemoryCache(),
  link: authLink.concat(link),
});

export default client;
