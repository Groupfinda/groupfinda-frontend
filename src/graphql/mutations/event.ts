import gql from "graphql-tag";

export type CreateEventData = {
  id: string;
};

export type CreateEventVariables = {
  title: string;
  description: string;
  dateOfEvent: Date;
  recurringMode: boolean;
  dateLastRegister: Date;
  images: string[];
  private: boolean;
  groupSize: number;
  category: string[];
  locationOn: boolean;
  location: {
    postalCode: string;
    address: string;
  };
};
export const CREATE_EVENT = gql`
  mutation createEvent(
    $title: String!
    $description: String!
    $dateOfEvent: Date!
    $recurringMode: Boolean!
    $dateLastRegister: Date!
    $images: [String]!
    $private: Boolean!
    $groupSize: Int!
    $category: [String]!
    $locationOn: Boolean!
    $location: LocationInput!
  ) {
    createEvent(
      title: $title
      description: $description
      dateOfEvent: $dateOfEvent
      recurringMode: $recurringMode
      dateLastRegister: $dateLastRegister
      images: $images
      private: $private
      groupSize: $groupSize
      category: $category
      locationOn: $locationOn
      location: $location
    ) {
      id
    }
  }
`;
