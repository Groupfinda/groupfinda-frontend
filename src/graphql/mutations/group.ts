import gql from "graphql-tag";

export type SendMessageData = boolean;
export type SendMessageVariables = {
  messageRoomId: string;
  message: string;
};
export const SEND_MESSAGE = gql`
  mutation sendMessage($messageRoomId: ID!, $message: MessageInput!) {
    sendMessage(messageRoomId: $messageRoomId, message: $message)
  }
`;
