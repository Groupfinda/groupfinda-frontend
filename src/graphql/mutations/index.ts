export * from "./user";
export * from "./event";
export * from "./profile";
export * from "./util";
export * from "./group";
