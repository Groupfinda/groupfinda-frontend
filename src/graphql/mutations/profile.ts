import gql from "graphql-tag";

export const SAVE_ANSWER = gql`
    mutation submitRangeQuestion(
        $order: Int!
        $value: Int!
    ) {
        submitRangeQuestion(
            order: $order
            value: $value
        )
    }
`;

export const UPDATE_PROFILE = gql `
    mutation updateProfileField(
        $userHobbies: [String]
        $userFaculty: String
        $userYearOfStudy: Int
    ) {
        updateProfileField(
            userHobbies: $userHobbies
            userFaculty: $userFaculty
            userYearOfStudy: $userYearOfStudy
        )
    }
`;