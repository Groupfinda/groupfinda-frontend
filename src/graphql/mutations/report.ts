import gql from "graphql-tag";

export const SUBMIT_REPORT = gql`
    mutation submitReport (
        $title: String!
        $category: String!
        $description: String!
    ) {
        submitReport(
            title: $title
            category: $category
            description: $description
        )
    }
`;