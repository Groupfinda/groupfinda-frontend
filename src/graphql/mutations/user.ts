import gql from "graphql-tag";

export const LOGIN_USER = gql`
  mutation loginUser($username: String!, $password: String!) {
    loginUser(username: $username, password: $password) {
      token
    }
  }
`;

export const CREATE_USER = gql`
  mutation createUser(
    $username: String!
    $password: String!
    $confirmPassword: String!
    $firstName: String!
    $lastName: String!
    $email: String!
    $gender: String!
    $birthday: Date!
  ) {
    createUser(
      username: $username
      password: $password
      confirmPassword: $confirmPassword
      firstName: $firstName
      lastName: $lastName
      email: $email
      gender: $gender
      birthday: $birthday
    ) {
      token
    }
  }
`;

export const FORGET_PASSWORD = gql`
  mutation forgetPassword($username: String!, $email: String!) {
    forgetPassword(username: $username, email: $email)
  }
`;

export const RESET_PASSWORD = gql`
  mutation resetPassword(
    $originalPassword: String!
    $newPassword: String!
    $confirmNewPassword: String!
  ) {
    resetPassword(
      originalPassword: $originalPassword
      newPassword: $newPassword
      confirmNewPassword: $confirmNewPassword
    )
  }
`;

export const REFETCH_QUERY = gql`
  mutation {
    refetchQuery
  }
`;

export const UPDATE_USER = gql`
  mutation updateUserField(
    $firstName: String
    $lastName: String
    $birthday: Date
    $gender: String
    $avatar: String
    $lowerAge: Int
    $upperAge: Int
    $maxDistance: Int
  ) {
    updateUserField(
      firstName: $firstName
      lastName: $lastName
      birthday: $birthday
      gender: $gender
      avatar: $avatar
      lowerAge: $lowerAge
      upperAge: $upperAge
      maxDistance: $maxDistance
    )
  }
`;
