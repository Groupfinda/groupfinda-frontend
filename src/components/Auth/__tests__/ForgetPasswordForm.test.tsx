import React from "react";
import { render } from "../../../../utils/test-utils";
import ForgetPasswordForm from "../ForgetPasswordForm";
import { MockedResponse } from "@apollo/react-testing";
import { FORGET_PASSWORD } from "../../../graphql/mutations";
import { GraphQLError } from "graphql";

const mocks: MockedResponse[] = [
  {
    request: {
      query: FORGET_PASSWORD,
      variables: {
        username: "test1",
        email: "test1",
      },
    },
    result: {
      errors: [
        new GraphQLError(
          "Error 1",
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          {
            exception: { invalidArgs: ["username", "email"] },
          }
        ),
      ],
    },
  },
  {
    request: {
      query: FORGET_PASSWORD,
      variables: {
        username: "test2",
        email: "test2",
      },
    },
    result: {
      errors: [
        new GraphQLError(
          "Error 2",
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          {
            exception: { invalidArgs: ["username"] },
          }
        ),
      ],
    },
  },
  {
    request: {
      query: FORGET_PASSWORD,
      variables: {
        username: "test3",
        email: "test3",
      },
    },
    result: {
      errors: [
        new GraphQLError(
          "Error 3",
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          {
            exception: { invalidArgs: [, "email"] },
          }
        ),
      ],
    },
  },
];

test("renders UI correctly", () => {
  const tree = render(<ForgetPasswordForm />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders text components correctly", () => {
  const component = render(<ForgetPasswordForm />);
  expect(component.queryByText("Username")).toBeTruthy();
  expect(component.queryByPlaceholder("Username")).toBeTruthy();
  expect(component.queryByText("Email")).toBeTruthy();
  expect(component.queryByPlaceholder("Email")).toBeTruthy();
  expect(component.queryByText("Reset my password")).toBeTruthy();
  console.log(component.queryByProps({ children: "Reset my password" }));
});
