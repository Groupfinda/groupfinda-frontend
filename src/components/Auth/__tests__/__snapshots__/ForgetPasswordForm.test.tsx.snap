// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UI correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    appearance="default"
    level="1"
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
        },
        Object {
          "alignSelf": "stretch",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "marginBottom": 5,
        }
      }
    >
      <Text
        appearance="default"
        category="p1"
        style={
          Array [
            Object {
              "color": "#222B45",
              "fontFamily": "System",
              "fontSize": 15,
              "fontWeight": "400",
            },
            Array [
              Object {
                "color": "#8F9BB3",
                "fontFamily": "System",
                "fontSize": 12,
                "fontWeight": "800",
                "marginBottom": 4,
              },
              Object {
                "textAlign": "left",
              },
            ],
          ]
        }
      >
        Username
      </Text>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#F7F9FC",
              "borderColor": "#E4E9F2",
              "borderRadius": 4,
              "borderWidth": 1,
              "minHeight": 40,
              "paddingHorizontal": 8,
              "paddingVertical": 7,
            },
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "width": "100%",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={24}
            bbWidth={24}
            fill="#8F9BB3"
            focusable={false}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "height": 24,
                  "marginHorizontal": 8,
                  "tintColor": "#8F9BB3",
                  "width": 24,
                },
                Object {
                  "flex": 0,
                  "height": 24,
                  "width": 24,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
          >
            <RNSVGGroup
              fill={4287601587}
              propList={
                Array [
                  "fill",
                ]
              }
            >
              <RNSVGGroup>
                <RNSVGGroup>
                  <RNSVGRect
                    height="24"
                    opacity={0}
                    width="24"
                    x={0}
                    y={0}
                  />
                  <RNSVGPath
                    d="M12 11a4 4 0 1 0-4-4 4 4 0 0 0 4 4zm0-6a2 2 0 1 1-2 2 2 2 0 0 1 2-2z"
                  />
                  <RNSVGPath
                    d="M12 13a7 7 0 0 0-7 7 1 1 0 0 0 2 0 5 5 0 0 1 10 0 1 1 0 0 0 2 0 7 7 0 0 0-7-7z"
                  />
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <TextInput
          allowFontScaling={true}
          appearance="default"
          autoCapitalize="none"
          autoCorrect={false}
          blurOnSubmit={false}
          editable={true}
          onBlur={[Function]}
          onChange={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Username"
          placeholderTextColor="#8F9BB3"
          rejectResponderTermination={true}
          size="medium"
          status="basic"
          style={
            Array [
              Object {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "normal",
                "marginHorizontal": 8,
              },
              Object {
                "flexBasis": "auto",
                "flexGrow": 1,
                "flexShrink": 1,
              },
              null,
              undefined,
            ]
          }
          textContentType="username"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Array [
            Object {
              "marginTop": 4,
            },
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
          ]
        }
      />
    </View>
    <View
      style={Object {}}
    >
      <Text
        appearance="default"
        category="p1"
        style={
          Array [
            Object {
              "color": "#222B45",
              "fontFamily": "System",
              "fontSize": 15,
              "fontWeight": "400",
            },
            Array [
              Object {
                "color": "#8F9BB3",
                "fontFamily": "System",
                "fontSize": 12,
                "fontWeight": "800",
                "marginBottom": 4,
              },
              Object {
                "textAlign": "left",
              },
            ],
          ]
        }
      >
        Email
      </Text>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#F7F9FC",
              "borderColor": "#E4E9F2",
              "borderRadius": 4,
              "borderWidth": 1,
              "minHeight": 40,
              "paddingHorizontal": 8,
              "paddingVertical": 7,
            },
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "width": "100%",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={24}
            bbWidth={24}
            fill="#8F9BB3"
            focusable={false}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "height": 24,
                  "marginHorizontal": 8,
                  "tintColor": "#8F9BB3",
                  "width": 24,
                },
                Object {
                  "flex": 0,
                  "height": 24,
                  "width": 24,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
          >
            <RNSVGGroup
              fill={4287601587}
              propList={
                Array [
                  "fill",
                ]
              }
            >
              <RNSVGGroup>
                <RNSVGGroup>
                  <RNSVGRect
                    height="24"
                    opacity={0}
                    width="24"
                    x={0}
                    y={0}
                  />
                  <RNSVGPath
                    d="M19 4H5a3 3 0 0 0-3 3v10a3 3 0 0 0 3 3h14a3 3 0 0 0 3-3V7a3 3 0 0 0-3-3zm-.67 2L12 10.75 5.67 6zM19 18H5a1 1 0 0 1-1-1V7.25l7.4 5.55a1 1 0 0 0 .6.2 1 1 0 0 0 .6-.2L20 7.25V17a1 1 0 0 1-1 1z"
                  />
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <TextInput
          allowFontScaling={true}
          appearance="default"
          autoCapitalize="none"
          autoCorrect={false}
          editable={true}
          onBlur={[Function]}
          onChange={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Email"
          placeholderTextColor="#8F9BB3"
          rejectResponderTermination={true}
          size="medium"
          status="basic"
          style={
            Array [
              Object {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "normal",
                "marginHorizontal": 8,
              },
              Object {
                "flexBasis": "auto",
                "flexGrow": 1,
                "flexShrink": 1,
              },
              null,
              undefined,
            ]
          }
          textContentType="emailAddress"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Array [
            Object {
              "marginTop": 4,
            },
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
          ]
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#008EED",
          "borderColor": "#008EED",
          "borderRadius": 4,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 40,
          "minHeight": 40,
          "minWidth": 40,
          "opacity": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 12,
        }
      }
    >
      <Text
        appearance="default"
        category="p1"
        style={
          Array [
            Object {
              "color": "#222B45",
              "fontFamily": "System",
              "fontSize": 15,
              "fontWeight": "400",
            },
            Object {
              "color": "#FFFFFF",
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "bold",
              "marginHorizontal": 10,
            },
          ]
        }
      >
        Reset my password
      </Text>
    </View>
  </View>
</View>
`;
